### ISSUE TITLE
Test if a factory exists

### ISSUE THREAD

**Issue by laceysanderson** [statonlab/TripalTestSuite#92]
**How would I test if a factory exists before I use it in a test?**

For example, say I have a dynamic variable containing the chado table, which can be set based on the dataProvider. Since there are not yet factories for all chado tables, I would need to check if the factory exists before creating fake data using it...
```php
if (factory('chado.' . $chado_table)->exists()) {
  factory('chado.' . $chado_table, $num_records)->create();
}
```

**Comment by almasaeed2010** [comment]
When the factory doesn’t exist, an exception is thrown. So you can try catch? Otherwise we can add a static method to the Factory class that can check for you if you feel that’d be better.

Just let me know :)

**Comment by laceysanderson** [comment]
I personally prefer to check ahead of time rather then catch afterwards (in general) so a static method would be great! In my particular case, I just want to skip some assertions in the case where I can't generate fake data (no failures) so checking ahead just feels more correct?

**Comment by almasaeed2010** [comment]
You got it! I’ll release a new version in a few hours.

Thanks!

**Comment by almasaeed2010** [comment]
We've implemented the new method in 1.41 (composer update should do it). Please see [the docs](https://tripaltestsuite.readthedocs.io/en/latest/factories.html#checking-if-a-factory-exists).

Let me know if you have any questions.

Thanks!

**Comment by laceysanderson** [comment]
Works great! Thanks!!!
